### PR TITLE
Removes Map representation from protocol

### PR DESCRIPTION
**Pull request by sancar**
This change does not affect wire level protocol.

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java [hunk 1]
```diff
@@ -1361,21 +1362,21 @@ public void putAll(Map<? extends K, ? extends V> m) {
             Data keyData = toData(entry.getKey());
 
             int partitionId = partitionService.getPartitionId(keyData);
-            Map<Data, Data> partition = entryMap.get(partitionId);
+            List<Map.Entry<Data, Data>> partition = entryMap.get(partitionId);
             if (partition == null) {
-                partition = new HashMap<Data, Data>();
+                partition = new LinkedList<Entry<Data, Data>>();
```
Comment (LoneRifle) [inline hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java]: Given http://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist , would it make better sense to use ArrayList here? I get that if the backing array has to be resized we incur a time penalty, but that may be mitigated by estimating the size needed on construction.

**Comment by cangencer** [comment]
Does the code templates for other languages need updating also to support "MAPENTRY"?

**Comment by sancar** [comment]
Yes, they need to be updated. Since I removed "MAP", code generation will be broken for other languages. But I could not do it because I don't know what to put instead of new AbstractMap.Entry class.

File: hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java [hunk 2]
```diff
@@ -1361,21 +1361,21 @@ public void putAll(Map<? extends K, ? extends V> m) {
             Data keyData = toData(entry.getKey());
 
             int partitionId = partitionService.getPartitionId(keyData);
-            Map<Data, Data> partition = entryMap.get(partitionId);
+            List<Map.Entry<Data, Data>> partition = entryMap.get(partitionId);
             if (partition == null) {
-                partition = new HashMap<Data, Data>();
+                partition = new ArrayList<Entry<Data, Data>>();
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java]: is this supposed to be Map.Entry instead of just Entry?

**Comment by sancar** [comment]
run-lab-run

**Comment by cangencer** [comment]
:+1:

**Comment by sancar** [comment]
run-lab-run

**Pull request merged**